wallet/types/src/wallet_block.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode};
+
+use common::{
+    chain::{block::ConsensusData, Block, GenBlock, Genesis, TxInput, TxOutput},
+    primitives::{BlockHeight, Id, Idable},
+};
+
+#[derive(Debug, PartialEq, Eq, Clone, Decode, Encode)]
+pub struct WalletBlock {
```

Comment:
Please add documentation, explaining why this struct even needs to exist and what problem it solves.